web/src/people.ts

```diff
@@ -1470,6 +1470,17 @@ export function make_user(user_id: number, email: string, full_name: string): Us
     };
 }
 
+export function add_inaccessible_user(user_id: number): User {
+    if (people_by_user_id_dict.has(user_id)) {
+        return get_by_user_id(user_id);
+    }
+
+    const email = "user" + user_id + "@" + page_params.realm_bot_domain;
+    const unknown_user = make_user(user_id, email, INACCESSIBLE_USER_NAME);
+    _add_user(unknown_user);
```

Comment:
Do we want to add this user to `active_user_dict` as well? We do have `is_active` as `true` for all inaccessible users. Asking this as we do not do it when making fake user objects for message senders.

Reply:
I think it would basically be reasonable to just make a fake user using `add_inaccessible_user` whenever we are processing a `user_id` that the server assures us must exist (i.e. something the server gave us, not like a value put into a potentially bogus narrow URL). So I'm not sure why we wouldn't want to do this in the message sender code path and emoji author code paths too.

Reply:
I think it might make sense to abstract this more nicely with just having a `get_user_by_id_assert_valid()` function that calls `add_accessible_user` if the user doesn't exist -- a lot of these little commits, like the bots one, where we wouldn't necessarily need special logic, might then be just replacing `get_user_by_id` with `get_user_by_id_assert_valid` without broader changes.

Reply:
Ok, I got this. Will update the PR to create the fake user objects for all `user_id` provided by server.

But the question that I was asking above was that before this PR as well, we created fake user objects for the users who are sender of the messages received by client but the client has not received the data for them. In that case we do not add the user in `active_user_dict`, which we use at certain places to determine if the user is active or not.

So, do we want to change this behavior now to add the users in `active_user_dict` as well?

Reply:
Yeah I think that's the right plan.

Next year we'll do some API design discussions about reworking how "display recipients" work, which may involve changing the overall policy here, but I don't see any technical problem with the current approach.